Покрыт тестами класс Burger

DawieJohnson opened the pull request.

Commit:
Покрыт тестами класс Burger

okryukova commented on src/test/java/praktikum/BurgerEdgeCasesTest.java:
```diff
@@ -0,0 +1,44 @@
+    @Test
+    public void testBurgerInitialState() {
+        Burger burger = new Burger();
+        assertNull(burger.bun);
```
⛔️Нужно исправить. Для юнит-тестов придерживаемся подхода: один тест, значит одна проверка. Если очень хочется несколько проверок -- тогда используем softAssertions. Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/praktikum/BurgerEdgeCasesTest.java:
```diff
@@ -0,0 +1,44 @@
+
+    @Test
+    public void testGetPriceWithZeroCost() {
+        Bun bun = new Bun("free bun", 0f);
```
⛔️Нужно исправить. Нужно использовать моки

okryukova commented on src/test/java/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,168 @@
+        Burger testBurger = new Burger();
+        testBurger.setBuns(mockBun);
+
+        Ingredient ing1 = mock(Ingredient.class);
```
⛔️Нужно исправить. При нейминге не рекомендуется использовать числа (Field2), их еще называют magicNumbers. Очень тяжело поддерживать код с magicNumbers.

okryukova commented on src/test/java/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,168 @@
+    @Test
+    public void testAddIngredient() {
+        int initialSize = burger.ingredients.size();
+        Ingredient newIngredient = mock(Ingredient.class);
```
⛔️Нужно исправить. Если в тесте не нужна параметризация, его нужно вынести в отдельный класс

okryukova commented on src/test/java/praktikum/BurgerTest.java:
```diff
@@ -0,0 +1,168 @@
+        assertTrue(receipt.contains("filling"));
+        assertTrue(receipt.contains("Price:"));
+    }
+}
```
⛔️Нужно исправить. Приложи, пожалуйста, отчет

Commit:
1. В BurgerEdgeCasesTest.java:

Разделен testBurgerInitialState() на три отдельных теста:
testBurgerBunInitialState()
testBurgerIngredientsInitialStateNotNull()
testBurgerIngredientsInitialStateEmpty()

2.В testGetPriceWithZeroCost():
Заменены реальные объекты на моки.

3."Исправить magic numbers" в BurgerTest.java:
Убраны ing1, ing2, Field2 и т.д.
Используются понятные имена: mockBun, mockIngredient
В параметризованном тесте создаются моки в цикле

4.Созданы отдельные классы:
BurgerTest.java - параметризованные тесты
BurgerBasicTest.java - базовые тесты без параметризации
BurgerEdgeCasesTest.java - edge cases тесты

5.Прикреплен отчет index.html

Commit:
1. ✅ ЗАМЕЧАНИЕ 1 - "Один тест - одна проверка"
BurgerBasicTest.java - ✅ исправлено (разделены тесты перемещения)

BurgerEdgeCasesTest.java - ⚠️ технически соответствует (каждый тест проверяет один аспект)

Остальные - ✅ исправлено

2. ✅ ЗАМЕЧАНИЕ 2 - "Проверка строки рецепта целиком"
✅ Полностью исправлено во всех файлах

✅ Удалены тесты с contains()

✅ Добавлены тесты с полным сравнением

3. ✅ ЗАМЕЧАНИЕ 3 - "Непараметризованные тесты в параметризованном классе"
✅ Полностью исправлено

✅ BurgerTest.java содержит только параметризованные тесты

✅ Все непараметризованные тесты вынесены в отдельные классы

4. ✅ ЗАМЕЧАНИЕ 4 - Прикреплена папка site
